randovania/games/am2r/exporter/hint_namer.py

```diff
@@ -102,6 +102,7 @@ def format_guaranteed_resource(
         )
 
     def format_location_hint(self, game: RandovaniaGame, pick_hint: PickupHint, hint: Hint, with_color: bool) -> str:
+        assert hint.precision is not None
```

Comment:
I think errors like this need to be fixed properly or at least have the root cause marked with a FIXME comment. it is a good thing that mypy is alerting us to problems this one

Reply:
`Hint.precision` can be None during the generator, because it's assigned later. When it reaches the exporter it should never be None.
Probably case of needing a different type for during generator and afterwards, but that's very annoying code.

Reply:
I don't mind that particular case, I suppose. I'm more bothered by the possibility of it being None because it isn't a normal hint to begin with. The ambiguity between those cases is a lot more problematic imo